scripts/dist-git/remove_unused_sources

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/bash
+
+if [[ $# != 3 ]]; then
+    echo "invalid number of arguments"
+    exit 1
+fi
+
+package="$1"
+lookaside_dir="$2"
+git_dir="$3"
+
+pkg_git_dir="$git_dir/$package.git"
+pkg_dir="$lookaside_dir/$package"
+
+if [ ! -d "$pkg_dir" ]; then
+    echo "$pkg_dir is not a valid directory."
+    exit 1
+fi
+if [ ! -d "$pkg_git_dir" ]; then
+    echo "$pkg_git_dir is not a valid directory."
+    exit 1
+fi
+
+pushd /tmp > /dev/null
+
+git clone "$pkg_git_dir" 2> /dev/null
+cloned_dir=`basename "$package"`
+
+if [ -d "$cloned_dir" ]; then
+    pushd "$cloned_dir" > /dev/null
+else
+    popd > /dev/null
+    exit 2
+fi
+
+branches=()
+whitelist=()
+
+eval "$(git for-each-ref --shell --format='branches+=(%(refname))' refs/remotes/)"
+for branch in "${branches[@]##*/}"; do
+    git checkout "$branch" &> /dev/null
+    if [ -f sources ]; then
+        while read line; do
+	    ll=($line)
+	    whitelist+=(${ll[1]})
```

Comment:
Note that sources line can contain either the old format:
`64c54d43787339a7cced48390eb3e1d0  uucp-1.07.tar.gz`
or the new one:
`SHA512 (tar-1.30.tar.xz.sig) = 149bfb112aae05e1f40a26ab2570fef71cf736eac62bf3a1bc8df905270c745500fc82764867323e440c67b1dd7116f5b0e1fd1977be7f10b333d481da3355e8`

Note that the name `ll` is not really obvious :-) and more common (and portable) statement is: 
```bash
set -- $line
filename=$2
```

Reply:
plus `test $# -gt 2 && filename=$(echo "$filename" | sed -e "s/^(//" -e "s/)$//")`